gtda/homology/simplicial.py

```diff
@@ -612,3 +613,202 @@ def transform(self, X, y=None):
+        precision.
+
+    max_edge_length : float, optional, default: ``numpy.inf``
+        Upper bound on the maximum value of the Vietoris-Rips filtration
+        parameter. Points whose distance is greater than this value will
+        never be connected by an edge, and topological features at scales
+        larger than this value will not be detected.
+
+    infinity_values : float or None, default : ``None``
+        Which death value to assign to features which are still alive at
+        filtration value `max_edge_length`. ``None`` has the same behaviour
+        as `max_edge_length`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    See also
+    --------
+    ConsistentRescaling
+
+    Notes
+    -----
+    `Flagser <https://github.com/luetge/flagser>`_ is used as a C++ backend
+    for computing Flag complexes persistent homology. Python bindings rely on
+    the `pyflagser <https://github.com/giotto-ai/pyflagser>`_ package.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] D. Luetgehetmann, D. Govc, J. P. Smith, and R. Levi, "Computing \
+        persistent homology of directed flag complexes", Algorithms,
+        13(1), 2020.
+
+    """
+
+    _hyperparameters = {'directed': [bool, [True, False]],
+                        'max_entries': [int, None],
+                        'max_edge_length': [numbers.Number, None],
+                        'infinity_values_': [numbers.Number],
+                        '_homology_dimensions': [list, [int, (0, np.inf)]],
+                        'coeff': [int, (2, np.inf)]}
+
+    def __init__(self, homology_dimensions=(0, 1), directed=True, coeff=2,
+                 max_entries=-1, max_edge_length=np.inf, infinity_values=None,
+                 n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.directed = directed
+        self.coeff = coeff
+        self.max_entries = max_entries
+        self.max_edge_length = max_edge_length
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _flagser_diagram(self, X):
+        X[X >= self.max_edge_length] = self.max_edge_length
+
+        Xdgms = flagser(X[X[:, 0] != np.inf],
+                        min_dimension=self._min_homology_dimension,
+                        max_dimension=self._max_homology_dimension,
+                        directed=self.directed, coeff=self.coeff,
+                        approximation=self.max_entries)['dgms']
+
+        if 0 in self._homology_dimensions:
+            Xdgms[0] = Xdgms[0][:-1, :]  # Remove final death at np.inf
```

Comment:
A negligible detail: maybe `Xdgms[0].slice(0,-1)` ?

Reply:
This is everywhere. So same answer as above. Documentation and code needs to be consistently updated throughout the package.

Reply:
But I am also not sure why it is better?